cmake/webgpu.cmake

```diff
@@ -0,0 +1,54 @@
+# Specify the filename to search for
+set(FILENAME "gpu.h")
+
+get_filename_component(PROJECT_ROOT ${CMAKE_CURRENT_SOURCE_DIR} DIRECTORY)
+get_filename_component(PROJECT_ROOT ${PROJECT_ROOT} DIRECTORY)
+
+# Construct potential paths
+set(FILEPATH_CURRENT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}")
+set(FILEPATH_PROJECT_ROOT "${PROJECT_ROOT}/${FILENAME}")
+
+# Check if the file exists in the current directory
+if(EXISTS ${FILEPATH_CURRENT_DIR})
+    set(TARGET_FILE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+elseif(EXISTS ${FILEPATH_PROJECT_ROOT})
+    set(TARGET_FILE_PATH ${PROJECT_ROOT})
+else()
+    message(FATAL_ERROR "File ${FILENAME} not found in either ${CMAKE_CURRENT_SOURCE_DIR} or ${CMAKE_CURRENT_SOURCE_DIR}/../../")
+endif()
+
+include(FetchContent)
+
+set(FETCHCONTENT_BASE_DIR "${TARGET_FILE_PATH}/third_party/fetchcontent")
+set(WEBGPU_DIST_LOCAL_PATH "${TARGET_FILE_PATH}/third_party/local/WebGPU-distribution")
+
+if(USE_LOCAL_LIBS)
+  set(WEBGPU_DIST_GIT_REPO ${WEBGPU_DIST_LOCAL_PATH})
+  message(STATUS "Using local WebGPU distribution: ${WEBGPU_DIST_LOCAL_PATH}")
+else()
+  set(WEBGPU_DIST_GIT_REPO "https://github.com/eliemichel/WebGPU-distribution")
+endif()
+
+option(WEBGPU_TAG "WebGPU distribution tag to use")
+if (NOT WEBGPU_TAG)
+  set(WEBGPU_TAG "dawn")
+endif()
+message(STATUS "Using WebGPU distribution tag: ${WEBGPU_TAG}")
+
+if (WEBGPU_TAG STREQUAL "dawn")
```

Comment:
At the time I think I ran into some issues trying to pin to a specific commit in the dawn branch, probably mostly due to my cmake knowledge limitations (i could get it to pin to a commit hash on main, but had issues with commit hashes on the branch).

If you know of an easy way to do it, that might be preferable for reproducibility.

Reply:
Can you give me a couple of tags to test with? Was this working before?